docs/RationaleRTL-SV.md

```diff
@@ -105,6 +105,111 @@ declarations in limited ways:
  - Interface signals are allowed to be zero bits wide.  They are dropped from
    Verilog emission.
 
+### Endianness: operand ordering and internal representation
+
+Certain operations require ordering to be defined (i.e. `rtl.concat`,
+`rtl.array_concat`, and `rtl.array_create`). There are two places where this
+is relevant: in the MLIR assembly and in the MLIR C++ model.
+
+In MLIR assembly, operands are always listed MSB to LSB (big endian style):
+
+```mlir
+%msb = rtl.constant (0xEF : i8) : i8
+%mid = rtl.constant (0x7 : i4) : i4
+%lsb = rtl.constant (0xA018 : i16) : i16
+%result = rtl.concat %msb, %mid, %lsb : (i8, i4, i16) -> i28
+// %result is 0xEF7A018
+```
+
+**Note**: Integers are always big endian in both bit and byte ordering.
```

Comment:
A written integer literal isn't the same thing as an in-memory integer, and 0xA018 (as a literal) could be stored either big or little endian.  I would say that Integer literals are written in left-to-write lexical order, rather than that they are big endian.

Reply:
I agree, "left-to-right" though